cmd/dockerd/daemon_unix.go

```diff
@@ -9,17 +9,38 @@ import (
 	"os/signal"
 	"path/filepath"
 	"strconv"
+	"strings"
 
 	"github.com/containerd/containerd/runtime/v1/linux"
 	"github.com/docker/docker/cmd/dockerd/hack"
 	"github.com/docker/docker/daemon"
 	"github.com/docker/docker/daemon/config"
 	"github.com/docker/docker/libcontainerd/supervisor"
+	"github.com/docker/docker/rootless"
 	"github.com/docker/libnetwork/portallocator"
 	"golang.org/x/sys/unix"
 )
 
-const defaultDaemonConfigFile = "/etc/docker/daemon.json"
+var (
+	defaultDaemonConfigDir  = "/etc/docker"
+	defaultDaemonConfigFile = "/etc/docker/daemon.json"
+)
+
+func init() {
+	if !rootless.RunningWithNonRootUsername {
+		return
+	}
+	// NOTE: CLI uses ~/.docker while the daemon uses ~/.config/docker, because
```

Comment:
This is duplicated, shouldn't it be in a function/package so that we can reuse it ? :angel:

Reply:
Sorry which line is duplicated? grepepd but could not find

Reply:
@AkihiroSuda like [here](https://github.com/moby/moby/pull/38050/files#diff-adcf97f8e8b7815a2ec4660c286531eaR28). They are very similar, I really think we could create a package for those :angel: